src/lib.rs

```diff
@@ -1,11 +1,20 @@
 use ark_ec::CurveGroup;
+use ark_ff::PrimeField;
 use ark_std::{fmt::Debug, rand::RngCore};
 
+pub mod transcript;
+
 #[derive(Debug)]
 pub enum Error {}
 
-pub trait Transcript<C: CurveGroup>: Debug {
-    // todos
+pub trait Transcript<F: PrimeField> {
+    type TranscriptConfig: Debug;
+
+    fn new(config: &Self::TranscriptConfig) -> Self;
```

Comment:
here I was not sure if to use `new` or `setup`

Reply:
I think `new` should be good